app/src/backend/GPCService.ts

```diff
@@ -30,18 +30,34 @@ export default class GPCService {
         },
       ],
     });
-
-    const subCategoryId = subcategory?.subcategoryId;
-    const subSectorId = subcategory?.subsector?.subsectorId;
-    const sectorId = subcategory?.subsector?.sector?.sectorId;
-
-    if (!sectorId || !subSectorId || !subCategoryId) {
-      throw new createHttpError.BadRequest(
-        "Couldn't find sector/ subsector/ subcategory for given GPC reference number",
-      );
+    if (subcategory) {
+      return {
+        sectorId: subcategory?.subsector?.sector?.sectorId,
+        subSectorId: subcategory?.subsector?.subsectorId,
+        subCategoryId: subcategory?.subsector?.sector?.sectorId,
```

Comment:
You can leave out the question mark after subcategory in these lines as you have just checked that is truthy with the if statement you're in.